src/input.rs

```diff
@@ -0,0 +1,99 @@
+use super::*;
+
+/// Access to the steam input interface
+pub struct Input<Manager> {
+    pub(crate) input: *mut sys::ISteamInput,
+    pub(crate) _inner: Arc<Inner<Manager>>,
+}
+
+impl<Manager> Input<Manager> {
+    /// Init must be called when starting use of this interface.
+    /// if explicitly_call_run_frame is called then you will need to manually call RunFrame
+    /// each frame, otherwise Steam Input will updated when SteamAPI_RunCallbacks() is called
+    pub fn init(&self, explicitly_call_run_frame: bool) {
+        unsafe {
+            sys::SteamAPI_ISteamInput_Init(self.input, explicitly_call_run_frame);
+        }
+    }
+
+    /// Synchronize API state with the latest Steam Input action data available. This
+    /// is performed automatically by SteamAPI_RunCallbacks, but for the absolute lowest
+    /// possible latency, you call this directly before reading controller state.
+    /// Note: This must be called from somewhere before GetConnectedControllers will
+    /// return any handles
+    pub fn run_frame(&self) {
+        unsafe { sys::SteamAPI_ISteamInput_RunFrame(self.input, false) }
+    }
+
+    /// Returns a list of the currently connected controllers
+    pub fn get_connected_controllers(&self) -> Vec<sys::InputHandle_t> {
```

Comment:
Could we have a version of this that does not allocate? Taking a slice, asserting/panicking the size is at least STEAM_INPUT_MAX_COUNT, and returning a subslice reference instead of a Vec.

Reply:
Can you show an example?

Reply:
```rust
pub fn get_connected_controllers(&self, controllers: impl AsMut<[InputHandle_t]>) {
    let handles = controllers.as_mut();
    assert!(handles.len() >= STEAM_INPUT_MAX_COUNT);
    unsafe {
      sys::SteamAPI_ISteamInput_GetConnectedControllers(self.input, handles.as_mut_ptr());
    }
}
```